runtime/foucoco/src/lib.rs

```diff
@@ -921,6 +924,341 @@ parameter_types! {
 	pub Schedule: pallet_contracts::Schedule<Runtime> = Default::default();
 }
 
+use frame_support::{
+	log::{error, warn},
+	pallet_prelude::*,
+};
+use sp_std::vec::Vec;
+
+use pallet_contracts::chain_extension::{
+	ChainExtension,
+	Environment,
+	Ext,
+	InitState,
+	RetVal,
+	SysConfig,
+	// UncheckedFrom,
+};
+use sp_core::crypto::UncheckedFrom;
+
+#[derive(Default)]
+pub struct Psp22Extension;
+
+use runtime_common::chain_ext::*;
+pub(crate) type BalanceOfForChainExt<T> =
+	<<T as orml_currencies::Config>::MultiCurrency as orml_traits::MultiCurrency<
+		<T as frame_system::Config>::AccountId,
+	>>::Balance;
+
+impl<T> ChainExtension<T> for Psp22Extension
+where
+	T: SysConfig
+		+ orml_tokens::Config<CurrencyId = CurrencyId>
+		+ pallet_contracts::Config
+		+ orml_currencies::Config<MultiCurrency = Tokens, AccountId = AccountId>
+		+ orml_currencies_allowance_ext::Config,
+	<T as SysConfig>::AccountId: UncheckedFrom<<T as SysConfig>::Hash> + AsRef<[u8]>,
+{
+	fn call<E: Ext>(&mut self, mut env: Environment<E, InitState>) -> Result<RetVal, DispatchError>
+	where
+		E: Ext<T = T>,
+		<E::T as SysConfig>::AccountId: UncheckedFrom<<E::T as SysConfig>::Hash> + AsRef<[u8]>,
+	{
+		let func_id = env.func_id();
+
+		warn!("func_id : {}", func_id);
+
+		match func_id {
+			//transfer
+			1105 => {
+				let ext = env.ext();
+				let address = ext.address().clone();
+				let caller = ext.caller().clone();
+				let mut env = env.buf_in_buf_out();
+				let create_asset: (
+					OriginType,
+					u8,
+					[u8; 12],
+					[u8; 32],
+					T::AccountId,
+					BalanceOfForChainExt<T>,
+				) = env.read_as()?;
+				let (origin_id, type_id, code, issuer, account_id, balance) = create_asset;
+
+				let address_account;
+				if origin_id == OriginType::Caller {
+					address_account = caller;
+				} else {
+					address_account = address;
+				}
+
+				warn!("asset_id : {:#?}", type_id);
+				warn!("address_account : {:#?}", address_account);
+				warn!("account_id : {:#?}", account_id);
+				warn!("balance : {:#?}", balance);
+
+				let currency_id = try_from(type_id, code, issuer).unwrap_or(CurrencyId::Native);
+
+				let is_allowed_currency =
+					orml_currencies_allowance_ext::Pallet::<T>::is_allowed_currency(currency_id);
+				if !is_allowed_currency {
+					return Err(DispatchError::Other(
+						"Currency id is not allowed for chain extension",
+					))
+				}
+
+				let result = <orml_currencies::Pallet<T> as MultiCurrency<T::AccountId>>::transfer(
+					currency_id,
+					&address_account,
+					&account_id,
+					balance,
+				);
+
+				warn!("result : {:#?}", result);
```

Comment:
Maybe rename `result` to `transfer_result` and also use
```
				warn!("transfer_result: {:#?}", result);
```

Reply:
Done